src/ServicePulse.Host/vue/src/components/monitoring/EndpointListRow.vue

```diff
@@ -1,70 +1,65 @@
 });
 const shortName = computed(() => {
-  return isGrouped.value ? settings.endpoint.shortName : "";
+  return isGrouped.value ? (settings.endpoint as GroupedEndpoint).shortName : endpoint.value.name;
 });
 const router = useRouter();
 const supportsEndpointCount = ref();
 const { historyPeriod: selectedPeriod } = storeToRefs(monitoringHistoryPeriodStore);
 
-function navigateToEndpointDetails($event, endpointName) {
-  if ($event.target.localName !== "button") {
-    router.push({ name: "endpoint-details", params: { endpointName: endpointName }, query: { historyPeriod: selectedPeriod.value.pVal } });
-  }
+function navigateToEndpointDetails(endpointName: string) {
+  router.push({ name: "endpoint-details", params: { endpointName: endpointName }, query: { historyPeriod: selectedPeriod.value.pVal } });
 }
 
-function formatGraphDuration(input) {
+function formatGraphDuration(input: EndpointValues) {
   if (input) {
     const lastValue = input.points.length > 0 ? input.points[input.points.length - 1] : 0;
     return useFormatTime(lastValue);
   }
-  return input;
+  return { value: "0", unit: "" };
 }
-function formatGraphDecimal(input, deci) {
-  if (input) {
-    const lastValue = input.points.length > 0 ? input.points[input.points.length - 1] : 0;
-    let decimals = 0;
-    if (lastValue < 10 || input > 1000000) {
-      decimals = 2;
-    }
-    return useFormatLargeNumber(lastValue, deci || decimals);
-  } else {
-    input = {
-      points: [],
-      average: 0,
-      displayValue: 0,
-    };
-    return input;
+
+function formatGraphDecimal(input: EndpointValues, deci: number) {
+  input = input ?? {
+    points: [],
+    average: 0,
+  };
+  const lastValue = input.points.length > 0 ? input.points[input.points.length - 1] : 0;
+  let decimals = 0;
+  if (lastValue < 10 || lastValue > 1000000) {
+    decimals = 2;
   }
+  return useFormatLargeNumber(lastValue, deci || decimals);
 }
 </script>
 
 <template>
   <div class="table-first-col endpoint-name name-overview">
     <div class="box-header">
       <div class="no-side-padding lead righ-side-ellipsis endpoint-details-link">
-        <a @click="navigateToEndpointDetails($event, endpoint.name)" class="cursorpointer" v-tooltip :title="endpoint.name">
-          {{ isGrouped ? shortName : endpoint.name }}
+        <a @click="navigateToEndpointDetails(endpoint.name)" class="cursorpointer" v-tooltip :title="endpoint.name">
+          {{ shortName }}
         </a>
-        <span class="endpoint-count" v-if="endpoint.connectedCount || endpoint.disconnectedCount" v-tooltip :title="`Endpoint instance(s):` + endpoint.connectedCount || 0">({{ endpoint.connectedCount || 0 }})</span>
+        <span class="endpoint-count" v-if="endpoint.connectedCount || endpoint.disconnectedCount" v-tooltip :title="`Endpoint instance(s):${endpoint.connectedCount || 0}`">({{ endpoint.connectedCount || 0 }})</span>
```

Comment:
I looked at https://github.com/Particular/ServicePulse/pull/1741/files#diff-5e11e94635a6c7dcf3c5e304785a3ae820793a350fa890e7e3c2b77927db7de7R10 and is not marked as nullable?

Reply:
removed || to 0